SU-HSDO/suhumsci
# STN-377: Utility Classes for Table Column Widths

STN-377

# READY FOR REVIEW

## Summary
This pull request adds a set of responsive utility classes to set the width of columns in table views. The naming convention for the table classes begins with `tb-table-col-` and then follows the same naming convention as the [Decanter Grid System](https://decanter.stanford.edu/page/layouts-grid-system/):

- `hb-table-col-3-of-12` sets the column width to 25% at all screen sizes
- `hb-table-col-md-3-of-12` sets the column width to 25% at the medium break point and larger

## Need Review By (Date)
05/20

## Urgency
Medium

## Steps to Test
1. Confirm that all tests pass
1. Modify a table pattern view and apply the following classes to a single field: `hb-table-col-6-of 12 hb-table-col-md-3-of-12`. I suggest editing: http://economics.suhumsci.loc/undergraduatehonors/honors-prize-and-award-winners
    <img width="1767" alt="Custom_Econ_Honors_Theses__Content____Department_of_Economics" src="https://user-images.githubusercontent.com/2486846/82339314-e483d300-99bb-11ea-80eb-65e6da2d2ec7.png">
    <img width="1791" alt="Custom_Econ_Honors_Theses__Content____Department_of_Economics" src="https://user-images.githubusercontent.com/2486846/82339489-1432db00-99bc-11ea-8aa0-596a088c5344.png">
    This should cause the first table column to be 50%  at the smallest screen sizes then change to 25% at the medium breakpoint.
    ![BeFunky-design (1)](https://user-images.githubusercontent.com/2486846/82345230-06cd1f00-99c3-11ea-8d10-e0270dc7cf5d.jpg)

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
